api/types/authentication.go

```diff
@@ -402,9 +416,13 @@ func (c *AuthPreferenceV2) CheckAndSetDefaults() error {
 		c.SetOrigin(OriginDynamic)
 	}
 
-	// make sure type makes sense
 	switch c.Spec.Type {
-	case constants.Local, constants.OIDC, constants.SAML, constants.Github:
+	case constants.Local:
+		if !c.Spec.AllowLocalAuth.Value {
+			log.Warn("Ignoring local_auth=false when authentication.type=local")
```

Comment:
if this is a misconfiguration should we return an error rather than just continuing? Or would that break backwards-compatibility

Reply:
Yep, I didn't error because I was concerned about backwards compatibility. To be fair we didn't even need this change, it's just something I noticed while adding the other flags.